### PR TITLE
Add ASR recipe of MAGICDATA mandarin read speech

### PR DESCRIPTION
**Pull request by tjysdsg**
http://www.openslr.org/68/

Most of the code is copied from aishell2 recipe

TODOs:

- [x] Store training results in README.md
    - [x] Preliminary results
    - [x] Fix training loss "hill climb"
- [x] Automatically download data and unzip

### CODE REVIEW

File: egs2/magicdata/asr1/local/data.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    # Normalize text to capital letters
+    for x in train dev test; do
+        mv data/${x}/text data/${x}/text.org
+        paste <(cut -f 1 data/${x}/text.org) <(cut -f 2 data/${x}/text.org | tr '[:lower:]' '[:upper:]') \
+            > data/${x}/text
+        rm data/${x}/text.org
+    done
+
+    echo "Exclude train utterances with English tokens. "
```
Comment (ftshijt) [inline egs2/magicdata/asr1/local/data.sh:56]: How about making it into two separate stages

File: egs2/magicdata/asr1/run.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    --lm_config "${lm_config}"                         \
+    --asr_config "${asr_config}"                       \
+    --inference_config "${inference_config}"           \
+    --train_set "${train_set}"                         \
+    --valid_set "${valid_set}"                         \
+    --test_sets "${test_sets}"                         \
+    --speed_perturb_factors "${speed_perturb_factors}" \
+    --asr_speech_fold_length 512 \
+    --asr_text_fold_length 150 \
+    --lm_fold_length 150 \
```
Comment (ftshijt) [inline egs2/magicdata/asr1/run.sh:40]: any special reason to change the default value?

**Comment by tjysdsg** [comment]
@ftshijt Hi Jiatong, thanks for the review. I'll fix these issues in future commits and train a model using the new configuration

File: egs2/magicdata/asr1/run.sh (hunk 2 above)
Comment (tjysdsg) [reply egs2/magicdata/asr1/run.sh]: Resolved in de23a67dccfd2af3fc95fd50020a03130f6a0dc1

File: egs2/magicdata/asr1/conf/tuning/transducer/train_conformer-rnn_transducer.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
+# Trained with Tesla V100 (32GB) x 8 GPUs. It takes about 4 days.
```
[the remark this answers is not in the document]
Comment (tjysdsg) [reply egs2/magicdata/asr1/conf/tuning/transducer/train_conformer-rnn_transducer.yaml]: Resolved in 2713f7c264a5dda1dda9517ca805ea8f4203f7d1

File: egs2/magicdata/asr1/conf/train_asr_conformer.yaml [hunk 4]
```diff
@@ -0,0 +1 @@
+tuning/train_asr_conformer.yaml
```
[the remark this answers is not in the document]
Comment (tjysdsg) [reply egs2/magicdata/asr1/conf/train_asr_conformer.yaml]: Resolved in 2713f7c264a5dda1dda9517ca805ea8f4203f7d1

**Comment by ftshijt** [comment]
If you think the PR is ready, you can remove the WIP tag in the title. And I can proceed to a final review.

**Comment by tjysdsg** [comment]
> If you think the PR is ready, you can remove the WIP tag in the title. And I can proceed to a final review.

The PR is not ready yet, since the training loss of the second model I trained yesterday goes up significantly after 7epoch. The 7-th epoch does seem to show better performance than the first model I trained.

Now the cluster is back online, I'm training another one with a new learning rate and warmup_step. I'll let you know if the PR is ready for review. Thanks

File: egs2/magicdata/asr1/local/data.sh (hunk 1 above)
Comment (tjysdsg) [reply egs2/magicdata/asr1/local/data.sh]: Resolved in 2a188ecb7af5f929ade4403f0005a9a3cda53aec

**Comment by tjysdsg** [comment]
@ftshijt Hi Jiatong, this PR is now ready for review. Thanks!

File: egs2/magicdata/asr1/README.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+## Environments
+
+- date: `Mon Sep 19 20:16:36 EDT 2022`
+- python version: `3.9.12 (main, Apr  5 2022, 06:56:58)  [GCC 7.5.0]`
+- espnet version: `espnet 202207`
+- pytorch version: `pytorch 1.8.1+cu102`
+- Git hash: `30d08e87dfdfdc35ad6f438f01795c3efa531acf`
+    - Commit date: `Mon Sep 19 15:31:07 2022 -0400`
+
+## [Conformer](conf/tuning/train_asr_conformer.yaml) with [Transformer-LM](conf/tuning/train_lm_transformer.yaml)
```
Comment (ftshijt) [inline egs2/magicdata/asr1/README.md:12]: It would be great if you can also upload the model into Huggingface.

Comment (ftshijt) [reply egs2/magicdata/asr1/README.md:12]: You can finish that in another PR

**Review by ftshijt — COMMENTED** [review]
Many thanks for the contribution. LGTM! I will merge it once the CI is passed

File: egs2/magicdata/asr1/README.md (hunk 5 above)
Comment (tjysdsg) [reply egs2/magicdata/asr1/README.md:12]: Got it. I'll train a couple of more epochs just to be safe, upload the model to HF, and update README.md in another PR.

**Comment by tjysdsg** [comment]
@ftshijt Seems like the CI script itself could be broken? All recent commits to the master branch failed to pass CI.

**Comment by ftshijt** [comment]
Yeah, understand, we will have people working on fixing the CI. Please allow some time for it to be fixed.

**Comment by tjysdsg** [comment]
@ftshijt Hi, the model has been uploaded to HuggingFace. No further PR is needed :)

**Review by ftshijt — APPROVED** [review]
Everything seems to be good on my side. Will be wait the CI to be fixed

**Comment by ftshijt** [comment]
Can you merge the latest Commit to fix the CI issues?

**Comment by tjysdsg** [comment]
@ftshijt Hi, sorry for the late reply and thanks for merging it for me!
